src/app/content/components/Page/mediaModalManager.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { ReactNode, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import MediaModal from './MediaModal';
+import { HTMLElement, MouseEvent, KeyboardEvent, HTMLImageElement } from '@openstax/types/lib.dom';
+
+
+export function createMediaModalManager(container: HTMLElement | null) {
```

Comment:
This would be more readable if you could extract out the definitions for subunits like MediaModalPortal and create them with a single call (e.g., MediaModalPortal = createMediaModalPortal(mount)).
`handleMediaInteraction` and the listener functions could be defined in a hook.
The keydown functionality could be another hook.